Brew formula

dgreatwood opened the pull request:
This PR adds the Homebrew formula for Pistache to the Pistache source tree. This is the formula used by the popular brew package manager to install Pistache on macOS.

Updated files:
 - README.md: Updated to reflect the option of a brew install in macOS
 - Building on macOS.txt: Updated to reflect the option of a brew install
 - meson.build: Deal with differing howard-hinnant-date names (required by brew)
 - tests/meson.build: Allow use of subproject httplib.h with brew
 - rest_server_test.cc, rest_swagger_server_test.cc, router_test.cc: Suppress httplib.h GCC warning (causing build issue on Ubuntu). Also, removed some whitespace.

New files:
- pistache.rb: This is the brew formula
- runformula.sh: A script for developers, to install or test the Pistache brew formula
- brew.yaml: GitHub workflow to test the brew formula

kiplingw commented on README.md:
```diff
@@ -103,7 +107,7 @@ $ sudo apt update
-#### Ubuntu PPA (Stable)
+### Ubuntu PPA (Stable)
 
 Currently there are no stable release of Pistache published into the [stable](https://launchpad.net/~pistache+team/+archive/ubuntu/stable) PPA. However, when that time comes, run the following to install a stable package:
```
@dgreatwood, this isn't really an issue you created, but feel free to fix this paragraph to reflect the fact that we now have stable releases in the stable PPA.

dgreatwood commented:
Updated README and pushed to the same PR


On Thu, Dec 26, 2024 at 10:54 AM Kip ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/pistacheio/pistache/pull/1275#discussion_r1898064244>:
>
> > @@ -103,7 +107,7 @@ $ sudo apt update
>  $ sudo apt install libpistache-dev
>  ```
>
> -#### Ubuntu PPA (Stable)
> +### Ubuntu PPA (Stable)
>
>  Currently there are no stable release of Pistache published into the [stable](https://launchpad.net/~pistache+team/+archive/ubuntu/stable) PPA. However, when that time comes, run the following to install a stable package:
>
> @dgreatwood <https://github.com/dgreatwood>, this isn't really an issue
> you created, but feel free to fix this paragraph to reflect the fact that
> we now have stable releases in the stable PPA.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pistacheio/pistache/pull/1275#pullrequestreview-2523419556>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMA27HOYHIGH3VGGMASM32HRGFFAVCNFSM6AAAAABUHPLOFGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMRTGQYTSNJVGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
NOTICE: This email and its attachments may contain privileged and 
confidential information, only for the viewing and use of the intended 
recipient. If you are not the intended recipient, you are hereby notified 
that any disclosure, copying, distribution, acting upon, or use of the 
information contained in this email and its attachments is strictly 
prohibited and that this email and its attachments must be immediately 
returned to the sender and deleted from your system. If you received this 
email erroneously, please notify the sender immediately.  Xage Security, 
Inc. and its affiliates will never request personal information (e.g., 
passwords, Social Security numbers) via email.  Report suspicious emails to 
***@***.*** ***@***.***>

Tachi107 commented on .github/workflows/brew.yaml:
```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: 2022 Andrea Pappacoda
```
I didn't write this, I do not own copyright on the file :)

Tachi107 commented on meson.build:
```diff
@@ -36,6 +36,23 @@ if get_option('b_coverage')
 	add_project_arguments(compiler.get_supported_arguments(['-fstack-protector-all', '--param=ssp-buffer-size=4']), language: 'cpp')
 endif
 
+# howardhinnant/date has several names - look for them, from the most
+# to the least explicit name.
+# In Meson 0.60.0, this can be replaced with a simpler:
+#
+#     dependency('howardhinnant-date', 'hinnant-date', 'date')
+#
+date_dep = dependency('howard-hinnant-date', required: false)
+if not date_dep.found()
+	date_dep = dependency('howardhinnant-date', required: false)
+endif
+if not date_dep.found()
+	date_dep = dependency('hinnant-date', required: false)
+endif
+if not date_dep.found()
+	date_dep = dependency('date', fallback: ['hinnant-date', 'date_dep'])
+endif
+
```
Why is this here? It's the same code as below, and it's only needed if there's no working stdlib's C++20 chrono library.

dgreatwood replied:
Quite right on both counts... I updated.

Tachi107 reviewed (commented):
Two small comments :)

dgreatwood commented:
I'll look at these :-)

On Fri, Dec 27, 2024 at 6:37 AM Andrea Pappacoda ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Two small comments :)
> ------------------------------
>
> In .github/workflows/brew.yaml
> <https://github.com/pistacheio/pistache/pull/1275#discussion_r1898544129>:
>
> > @@ -0,0 +1,91 @@
> +# SPDX-FileCopyrightText: 2022 Andrea Pappacoda
>
> I didn't write this, I do not own copyright on the file :)
> ------------------------------
>
> In meson.build
> <https://github.com/pistacheio/pistache/pull/1275#discussion_r1898545984>:
>
> > +# howardhinnant/date has several names - look for them, from the most
> +# to the least explicit name.
> +# In Meson 0.60.0, this can be replaced with a simpler:
> +#
> +#     dependency('howardhinnant-date', 'hinnant-date', 'date')
> +#
> +date_dep = dependency('howard-hinnant-date', required: false)
> +if not date_dep.found()
> +	date_dep = dependency('howardhinnant-date', required: false)
> +endif
> +if not date_dep.found()
> +	date_dep = dependency('hinnant-date', required: false)
> +endif
> +if not date_dep.found()
> +	date_dep = dependency('date', fallback: ['hinnant-date', 'date_dep'])
> +endif
> +
>
> Why is this here? It's the same code as below, and it's only needed if
> there's no working stdlib's C++20 chrono library.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pistacheio/pistache/pull/1275#pullrequestreview-2524121690>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMA26T6JABQDMOC3OF4OT2HVQ3DAVCNFSM6AAAAABUHPLOFGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMRUGEZDCNRZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
NOTICE: This email and its attachments may contain privileged and 
confidential information, only for the viewing and use of the intended 
recipient. If you are not the intended recipient, you are hereby notified 
that any disclosure, copying, distribution, acting upon, or use of the 
information contained in this email and its attachments is strictly 
prohibited and that this email and its attachments must be immediately 
returned to the sender and deleted from your system. If you received this 
email erroneously, please notify the sender immediately.  Xage Security, 
Inc. and its affiliates will never request personal information (e.g., 
passwords, Social Security numbers) via email.  Report suspicious emails to 
***@***.*** ***@***.***>
